packages/amplify_datastore/android/src/main/java/com/amplifyframework/datastore/generated/model/AmplifyModelProvider.java

```diff
@@ -0,0 +1,53 @@
+package com.amplifyframework.datastore.generated.model;
+
+import com.amplifyframework.util.Immutable;
+import com.amplifyframework.core.model.Model;
+import com.amplifyframework.core.model.ModelProvider;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+/** 
+ *  Contains the set of model classes that implement {@link Model}
+ * interface.
+ */
+
+public final class AmplifyModelProvider implements ModelProvider {
+  private static final String AMPLIFY_MODEL_VERSION = "0d741308e0c992a62751cf9582613d1a";
+  private static AmplifyModelProvider amplifyGeneratedModelInstance;
+  private AmplifyModelProvider() {
+    
+  }
+  
+  public static AmplifyModelProvider getInstance() {
+    if (amplifyGeneratedModelInstance == null) {
+      amplifyGeneratedModelInstance = new AmplifyModelProvider();
+    }
+    return amplifyGeneratedModelInstance;
+  }
+  
+  /** 
+   * Get a set of the model classes.
+   * 
+   * @return a set of the model classes.
+   */
+  @Override
+   public Set<Class<? extends Model>> models() {
+    final Set<Class<? extends Model>> modifiableSet = new HashSet<>(
+          Arrays.<Class<? extends Model>>asList(Blog.class, Post.class, Comment.class, PostTest.class)
+        );
+    
+        return Immutable.of(modifiableSet);
+        
+  }
```

Comment:
Why do we have this in our main repo?

Reply:
Hey see the comments above - these were added for the sake of testing since we we didn't have a save method and I used the model builder methods directly instead to make nested models.  I have a note above to remove these before the actual merge.

Reply:
Jameson's example approach to writing nested serialized models is working as well but requires some extra steps / hard coding, so for the sake of easier testing I've kept them here for now.

Reply:
^ I use the same "mixed" approach in the Android library tests. Either save Java models, query serialized data -- or save serialized data, query Java models. I don't actually have a test right now to _both_ save and query serialized data.

Reply:
Java models have been removed.